File: app/src/main/java/ru/coffeeplanter/androidlogin/platform/LoginMocker.java
package ru.coffeeplanter.androidlogin.platform;

import android.os.Handler;

/**
 * Created by Ilya Solovyov on 18.09.2017.
 * is3k@ya.ru
 */

public class LoginMocker {

    private static final int DEFAULT_LOGIN_MOCK_TIME = 2000; // Delay in milliseconds.

    public static void execute(Runnable runnable) {
        execute(runnable, DEFAULT_LOGIN_MOCK_TIME);
    }

    public static void execute(Runnable runnable, int delay) {
        new Handler().postDelayed(runnable, delay);
    }

}

File: app/src/main/java/ru/coffeeplanter/androidlogin/platform/LoginMocker.kt
package ru.coffeeplanter.androidlogin.platform

import android.os.Handler

/**
 * Created by Ilya Solovyov on 18.09.2017.
 * is3k@ya.ru
 */

object LoginMocker {

    private val DEFAULT_LOGIN_MOCK_TIME = 2000 // Delay in milliseconds.

    fun execute(runnable: Runnable) {
        execute(runnable, DEFAULT_LOGIN_MOCK_TIME)
    }

    @Suppress("MemberVisibilityCanPrivate")
    fun execute(runnable: Runnable, delay: Int) {
        Handler().postDelayed(runnable, delay.toLong())
    }

}
